Assets/_BForBoss/_Core/Scripts/Boss/DerekBossManager.cs

```diff
@@ -31,55 +57,102 @@ public void UpdatePhase(DerekContextManager.Phase phase)
                 case DerekContextManager.Phase.FirstPhase:
                     //_healthBarViewBehavior.SetPhaseBoundary(0.75f, _onPhaseComplete.Invoke());
                     //_derekMissileBehavior.UpdateParameters(AdjustedParameters[])
-                    Debug.Log("Phase 1: Will transition at 75% health");
+                    Perigon.Utility.Logger.LogString("Phase 1: Will transition at 75% health", LoggerColor.Green, "derekboss");
                     break;
                 case DerekContextManager.Phase.SecondPhase:
                     //_healthBarViewBehavior.SetPhaseBoundary(0.25f, _onPhaseComplete.Invoke());
                     //_derekMissileBehavior.UpdateParameters(AdjustedParameters[])
-                    Debug.Log("Phase 2: Will transition at 25% health");
+                    Perigon.Utility.Logger.LogString("Phase 2: Will transition at 25% health", LoggerColor.Green, "derekboss");
                     break;
                 case DerekContextManager.Phase.FinalPhase:
                     //_healthBarViewBehavior.SetPhaseBoundary(0f, _onPhaseComplete.Invoke());
                     //_derekMissileBehavior.UpdateParameters(AdjustedParameters[])
-                    Debug.Log("Phase 3: Will transition at 0% health");
+                    Perigon.Utility.Logger.LogString("Phase 3: Will transition at 0% health", LoggerColor.Green, "derekboss");
                     break;
                 default:
-                    break;
+                    return;
             }
         }
 
         public void UpdateVulnerability(DerekContextManager.Vulnerability vulnerability)
         {
             switch (vulnerability)
             {
+                //Raise Shields
+                //Start Missile Launching
+                //Find Closest Death Wall and activate - 
+                //Start Movement with new given rotation (i.e. clockwise/anticlockwise)
                 case DerekContextManager.Vulnerability.Invulnerable:
-                    //Raise Shields
-                    //Recenter Derek Position
-                    //Find Closest Death Wall and activate - 
-                    //Start Movement with new given rotation (i.e. clockwise/anticlockwise)
-                    //Give small buffer of wait time - then StartMissile behavior
-                    //Ensure Derek is invincible in this case (edge case)
-                    
-                    Debug.Log("Raising Shields");
-                    Debug.Log("Resetting Position");
-                    Debug.Log("Activating Death Wall Closest to player");
-                    Debug.Log("Rotating to make sure death wall gets closer to player");
-                    Debug.Log("Starting Homing Missiles");
-                    Debug.Log("------------------------");
-                    
+                    _isVulnerable = false;
```

Comment:
so you have an `isInvulnerable` flag. but then also have the `Vulnerability` State. 

Can you only store one state? instead of having 2 sources of truth? `DerekContextManager.Vulnerability.Invulnerable` vs isVulnerable

Reply:
Good point, changed the var to be `DerekContextManager.Vulnerability` instead of a bool. It should be noted that this var is only being used at the moment for the temp Damage Receiving logic I have in the update loop, and will be removed as soon as the HealthBehaviour is implemented!

Added the todo comments in the cs to better clarify that